src/k8s/pkg/k8sd/api/cluster.go

```diff
@@ -13,27 +13,21 @@ import (
 )
 
 func (e *Endpoints) getClusterStatus(s *state.State, r *http.Request) response.Response {
-	snap := e.provider.Snap()
+	// fail if node is not initialized yet
+	if !s.Database.IsOpen() {
+		return response.Unavailable(fmt.Errorf("daemon not yet initialized"))
```

Comment:
seems like `Unavailable` is still in the 5xx space (503) so this shouldn't change anything on charm

Reply:
This is trying to imitate https://github.com/canonical/microcluster/blob/5043bc984c83e9ecfdba1de731ec05678b166f13/internal/rest/rest.go#L187-L196 as close as possible